addons/portal/static/src/js/portal_security.js

```diff
@@ -0,0 +1,266 @@
+            args: [session.user_id],
+        }));
+        const self = this;
+        const d_description = new Dialog(self, {
+            title: _t('New API Key'),
+            $content: qweb.render('portal.keydescription'),
+            buttons: [{text: _t('Confirm'), classes: 'btn-primary', close: true, click: async () => {
+                var description = d_description.el.querySelector('[name="description"]').value;
+                var wizard_id = await this._rpc({
+                    model: 'res.users.apikeys.description',
+                    method: 'create',
+                    args: [{name: description}],
+                });
+                var res = await handleCheckIdentity(
+                    this.proxy('_rpc'),
+                    this._rpc({
+                        model: 'res.users.apikeys.description',
+                        method: 'make_key',
+                        args: [wizard_id],
+                    })
+                );
+                const d_show = new Dialog(self, {
+                    title: _t('API Key Ready'),
+                    $content: qweb.render('portal.keyshow', {key: res.context.default_key}),
+                    buttons: [{text: _t('Close'), clases: 'btn-primary', close: true}],
+                });
+                d_show.on('closed', this, () => {
+                    window.location = window.location;
+                });
+                d_show.open();
+            }}, {text: _t('Discard'), close: true}],
+        });
+        d_description.opened(() => {
+            const input = d_description.el.querySelector('[name="description"]');
+            input.focus();
+            d_description.el.addEventListener('submit', (e) => {
+                e.preventDefault();
+                d_description.$footer.find('.btn-primary').click();
+            });
+        });
+        d_description.open();
+    }
+});
+
+publicWidget.registry.RemoveAPIKeyButton = publicWidget.Widget.extend({
+    selector: '.o_portal_remove_api_key',
+    events: {
+        click: '_onClick'
+    },
+
+    async _onClick(e){
+        e.preventDefault();
+        await handleCheckIdentity(
+            this.proxy('_rpc'),
+            this._rpc({
+                model: 'res.users.apikeys',
+                method: 'remove',
+                args: [parseInt(this.el.id)]
+            })
+        );
+        window.location = window.location;
+    }
+});
+
+publicWidget.registry.portalSecurity = publicWidget.Widget.extend({
+    selector: '.o_portal_security_body',
+
+    /**
+     * @override
+     */
+    init: function () {
+        // Show the "deactivate your account" modal if needed
+        $('.modal.show#portal_deactivate_account_modal').removeClass('d-block').modal('show');
+
+        // Remove the error messages when we close the modal,
+        // so when we re-open it again we get a fresh new form
+        $('.modal#portal_deactivate_account_modal').on('hide.bs.modal', (event) => {
```

Comment:
Same in this block, some remaining jquery to remove

Reply:
@sdegueldre could you help on what is the proper jQuery-less version of this one?

Reply:
As far as I'm aware, *if* jQuery is present on the page and we don't specifically opt-out, Bootstrap dispatches its custom events using jQuery meaning a non-jQuery event listener won't be receive them, so unfortunately there is no jQ-less version of this as long as we have jQuery (unless we opt out which would require converting all event listeners)

https://getbootstrap.com/docs/5.3/getting-started/javascript/#jquery-events